src/wp-includes/class-wp-query.php

```diff
@@ -1344,6 +1344,38 @@ public function parse_tax_query( &$q ) {
 		do_action( 'parse_tax_query', $this );
 	}
 
+
+	/**
+	 * Load comments from cache where possible, else make query and cache.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @global wpdb $wpdb WordPress database abstraction object.
+	 *
+	 * @param string $comments_request SQL query.
+	 * @return WP_Comment[] Array of `WP_Comment` objects.
+	 */
+	protected function get_comments( $comments_request ) {
```

Comment:
This method needs quite the refactor

* extenders can pass any SQL statement to the method and it's run without `prepare` and any other validation.
* WPCS sniffs won't pick this up as they're only set up for wpdb methods 
* There's not guarantee the query will actually return IDs of comments & store them in the comments cache group: `SELECT term_ids FROM wp_terms`

I'd recommend repeated code around the SELECTs below rather than prematurely trying to optimize for DRY.

Reply:
Removed dry